rechunker/api.py

```diff
@@ -10,12 +10,100 @@
 from rechunker.algorithm import rechunking_plan
 
 
-def rechunk_zarr2zarr_w_dask(
+def _shape_dict_to_tuple(dims, shape_dict):
+    # convert a dict of shape
+    shape = [shape_dict[dim] for dim in dims]
+    return tuple(shape)
+
+
+def _get_dims_from_zarr_array(z_array):
+    # use Xarray convention
+    # http://xarray.pydata.org/en/stable/internals.html#zarr-encoding-specification
+    return z_array.attrs["_ARRAY_DIMENSIONS"]
+
+
+def _zarr_empty(shape, store_or_group, chunks, dtype, name=None):
+    # wrapper that maybe creates the array within a group
+    if name is not None:
+        assert isinstance(store_or_group, zarr.hierarchy.Group)
```

Comment:
Debugging message

```suggestion
        msg = "store_or_group must be a zarr.heirarchy.Group when name is given"
        assert isinstance(store_or_group, zarr.hierarchy.Group), msg
```

That will probably fail